BootloaderCorePkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -379,6 +405,44 @@ AddSmbiosType (
 }
 
 
+/**
+  This function is called to initialize the SmbiosStringsPtr.
+**/
+VOID
+EFIAPI
+InitSmbiosStringPtr (
+  VOID
+  )
+{
+
+  EFI_STATUS            Status;
+  UINT16                Length;
+  VOID                 *SmbiosStringsPtr;
+
+  //
+  // Allocate the memmory for SMBIOS String Ptr,
+  //
+
+  SmbiosStringsPtr = AllocatePool ((PcdGet16(PcdSmbiosStringsCnt) * sizeof (SMBIOS_TYPE_STRINGS)) );
+  if (SmbiosStringsPtr == NULL) {
+    ASSERT_EFI_ERROR(EFI_OUT_OF_RESOURCES);
+  }
+
+  //
+  // Copy contents from Default String Array
+  //
+  Length = sizeof (mDefaultSmbiosStrings) / sizeof (SMBIOS_TYPE_STRINGS);
+  CopyMem (SmbiosStringsPtr, mDefaultSmbiosStrings, Length);
+
+  //
+  // Initialize SMBIOS String Ptr, Update Length
+  //
+  Status = PcdSet32S (PcdSmbiosStringsPtr, (UINT32)(UINTN)SmbiosStringsPtr);
+  ASSERT_EFI_ERROR (Status);
+  Status = PcdSet16S (PcdSmbiosStringsCnt, Length);
+  ASSERT_EFI_ERROR (Status);

```

Comment:
PcdSmbiosStringsCnt is the total count the SMBIOS table could have. It could be used when appending new items. 
why we update its value here?

Reply:
Yes, I should not update the PcdSmbiosStringsCnt.   Please remove #442 and #443.